From a9bac69559e24c61f989dc624028a996c7b47b09 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Mon, 17 Sep 2018 12:39:24 +0300
Subject: [PATCH 218/741] sdk_dpaa: ls1043a errata: align skb_shinfo

commit 750f6ad1def6a41109f48c950cb450f2aeaa4087 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Make sure the skb shared info is cache-line aligned when realigning
egress frames.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
index e1e5dfd485d3..0c5d89300d28 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
@@ -830,8 +830,10 @@ static struct sk_buff *a010022_realign_skb(struct sk_buff *skb,
 
 	/* For the new skb we only need the old one's data (both non-paged and
 	 * paged). We can skip the old tailroom.
+	 *
+	 * Make sure the skb_shinfo is cache-line aligned.
 	 */
-	nsize = headroom + skb->len +
+	nsize = SMP_CACHE_BYTES + DPA_SKB_SIZE(headroom + skb->len) +
 		SKB_DATA_ALIGN(sizeof(struct skb_shared_info));
 
 	/* Reserve enough memory to accommodate Jumbo frames */
-- 
2.17.1

